### PR TITLE
feat(nuxi): load `.env` when previewing builds

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/3621

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This loads `.env` in `nuxi preview`.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.docs/1.usage/8.cli.md [hunk 1]
```diff
@@ -67,6 +67,10 @@ Option        | Default          | Description
 
 This command sets `process.env.NODE_ENV` to `production`. To override, define `NODE_ENV` in a `.env` file or as command-line argument.
 
+::alert{type=info}
+For convenience, in preview mode, your `.env.production` or `.env` file will be loaded into `process.env`. (However, in production you will need to ensure your environment variables are set yourself.)
```
Comment (pi0) [inline docs/content/3.docs/1.usage/8.cli.md:71]: Do we support `.env.production` ?

File: packages/nuxi/src/commands/preview.ts [hunk 2]
```diff
@@ -34,6 +35,14 @@ export default defineNuxtCommand({
       process.exit(1)
     }
 
+    for (const env of ['.env.production', '.env']) {
```
Comment (pi0) [inline packages/nuxi/src/commands/preview.ts:38]: This is an anti pattern for supprting different names of `.env`

File: packages/nuxi/src/commands/preview.ts [hunk 3]
```diff
@@ -34,6 +35,14 @@ export default defineNuxtCommand({
...
+      if (existsSync(resolve(rootDir, env))) {
+        consola.info(`Loading \`${env}\` for preview...`)
+        process.env = await loadDotenv({ cwd: rootDir, fileName: env, env: process.env })
```
Comment (pi0) [inline packages/nuxi/src/commands/preview.ts:41]: WOuld be nice to somehow detect if file is loaded and make a warn it is only working for preview.

**Review by pi0 — APPROVED** [review]

**Comment by danielroe** [comment]
You should use runtimeConfig if you are consuming environment variables anywhere outside your Nuxt config file.

**Pull request merged**